### DISCUSSION TITLE
Improve logging with per job log file

### DISCUSSION THREAD

**Discussion by lluiscab** [mend/renovate-ce-ee#358] [Ideas]
I find myself trying to debug configuration issues by viewing the logs with `docker logs renovate --tail 50 -f`, while doing so, it's rather hard to see exactly when the job I'm interested in start or ends, and If there's a queue, the log stream will just go on and on if I don't stop it. 

It would be really useful if `renovate-ce` itself had a log directory where it saved a log file for each job, it could, for example, save files in a format `<log directory/<repo>/<jobId>.log>`.

Given that what `renovate-ce` does is basically running `renovate` and forwarding it's output to stdout, I don't think implementing such a feature would be _too difficult_. 

Is this something we could see implemented?

**Comment by rarkins** [comment]
Another way to do this would be to use the LOG_FILE approach combined with either a timestamp, repo name, and/or job ID

**Reply by lluiscab** [reply]
If there could be a way such that `renovate-ce` will pass a `LOG_FILE` parameter to `renovate` with a value generated from a template, that would probably work as well. Something like the following:

`MEND_RNV_LOG_FILE_TEMPLATE=logs/{repo}/{jobId}.log` that is passed to `renovate` as `LOG_FILE=logs/example-repo/1234.log`

**Reply by Gabriel-Ladzaretti** [reply]
`MEND_RNV_LOG_HISTORY_DIR`[^1] implements exactly that😄.

[^1]: https://github.com/mend/renovate-ce-ee/blob/main/docs/configure-renovate-ce-github.md

**Reply by lluiscab** [reply]
Yeah, the changes introduced in 6.4.0 seem to pretty much be all I needed, I'm currently waiting on https://github.com/mend/renovate-ce-ee/issues/365 to be resolved (in 6.6.0 I'd expect) in order to update and test this changes
